src/badguy/flame.cpp

```diff
@@ -56,25 +57,19 @@ Flame::Flame(const ReaderMapping& reader, int type) :
   {
     m_col.m_bbox.set_pos(Vector(m_start_position.x + cosf(angle) * radius,
                                 m_start_position.y + sinf(angle) * radius));
+    
   }
   m_countMe = false;
   m_glowing = true;
   SoundManager::current()->preload(FLAME_SOUND);
-
   set_colgroup_active(COLGROUP_TOUCHABLE);
-
-  switch (m_type)
-  {
-    case FIRE:
-      m_lightsprite->set_color(Color(0.21f, 0.13f, 0.08f));
-      break;
-    case GHOST:
-      m_lightsprite->set_color(Color(0.21f, 0.00f, 0.21f));
-      break;
-    case ICE:
-      m_lightsprite->set_color(Color(0.00f, 0.13f, 0.18f));
-      break;
+  std::vector<float> vColor;
+  if (reader.get("color", vColor)) {
+    lightcolor = Color(vColor);
```

Comment:
We don't use camelCase anywhere.
```suggestion
  std::vector<float> vcolor;
  if (reader.get("color", vcolor)) {
    lightcolor = Color(vcolor);
```